robot/framesystem/framesystem.go

```diff
@@ -279,15 +318,61 @@ func (svc *frameSystemService) TransformPointCloud(ctx context.Context, srcpc po
 }
 
 // PrefixRemoteParts applies prefixes to a list of FrameSystemParts appropriate to the remote they originate from.
-func PrefixRemoteParts(parts []*referenceframe.FrameSystemPart, remoteName, remoteParent string) {
+func PrefixRemoteParts(parts []*referenceframe.FrameSystemPart, remoteName, remoteParent string) []*PartConfig {
+	partConfigs := make([]*PartConfig, 0, len(parts))
 	for _, part := range parts {
-		if part.FrameConfig.Parent() == referenceframe.World { // rename World of remote parts
-			part.FrameConfig.SetParent(remoteParent)
+		if part.Origin.Parent() == referenceframe.World { // rename World of remote parts
+			part.Origin.SetParent(remoteParent)
 		}
 		// rename each non-world part with prefix
-		part.FrameConfig.SetName(remoteName + ":" + part.FrameConfig.Name())
-		if part.FrameConfig.Parent() != remoteParent {
-			part.FrameConfig.SetParent(remoteName + ":" + part.FrameConfig.Parent())
+		part.Origin.SetName(remoteName + ":" + part.Origin.Name())
+		if part.Origin.Parent() != remoteParent {
+			part.Origin.SetParent(remoteName + ":" + part.Origin.Parent())
+		}
+		partConfigs = append(partConfigs, &PartConfig{PreprocessedPart: part})
+	}
+	return partConfigs
+}
+
+// String prints out a table of each frame in the system, with columns of name, parent, translation and orientation.
+func (svc *frameSystemService) String() string {
+	t := table.NewWriter()
+	t.AppendHeader(table.Row{"#", "Name", "Parent", "Translation", "Orientation", "Geometry"})
+	t.AppendRow([]interface{}{"0", referenceframe.World, "", "", "", ""})
+	for i, part := range svc.parts {
+		pose := part.Origin.Pose()
+		tra := pose.Point()
+		ori := pose.Orientation().EulerAngles()
+		geomString := ""
+		if gc := part.Origin.Geometry(); gc != nil {
+			geomString = gc.String()
 		}
+		t.AppendRow([]interface{}{
+			fmt.Sprintf("%d", i+1),
+			part.Origin.Name(),
+			part.Origin.Parent(),
+			fmt.Sprintf("X:%.0f, Y:%.0f, Z:%.0f", tra.X, tra.Y, tra.Z),
+			fmt.Sprintf(
+				"Roll:%.2f, Pitch:%.2f, Yaw:%.2f",
+				utils.RadToDeg(ori.Roll),
+				utils.RadToDeg(ori.Pitch),
+				utils.RadToDeg(ori.Yaw),
+			),
+			geomString,
+		})
+	}
+	return t.Render()
+}
+
+// extractModelFrameJSON finds the robot part with a given name, checks to see if it implements ModelFrame, and returns the
+// JSON []byte if it does, or nil if it doesn't.
+func (svc *frameSystemService) extractModelFrameJSON(name string) (referenceframe.Model, error) {
+	part, ok := svc.components[name]
```

Comment:
if we're doing some renames, it might be nice to call this `robots` or `robotParts`?